### PR TITLE
feat: add org unit edit form

### PR DESCRIPTION
**Pull request by flaminic**

### CODE REVIEW

File: src/pages/organisationUnits/Edit.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    })
+
+    return (
+        <FormBase
+            onSubmit={useOnSubmitEdit({
+                section,
+                modelId,
+            })}
+            section={section}
+            initialValues={orgUnit.data as FormValues}
```
Comment (Birkbjo) [inline src/pages/organisationUnits/Edit.tsx:74]: Could we instead use `OrgUnitFormValues` as the type for the form. I think it is more correct than `FormValues` - which is the entire OrganisationUnit.

File: src/pages/organisationUnits/form/GeometryFields.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import i18n from '@dhis2/d2-i18n'
+import { Field, InputField } from '@dhis2/ui'
+import React from 'react'
+import { useField } from 'react-final-form'
+
+export function GeometryFields() {
```
Comment (Birkbjo) [inline src/pages/organisationUnits/form/GeometryFields.tsx:6]: I think we should add a type for `Geometry`. And we get it from `DefinitelyTyped`: https://www.npmjs.com/package/@types/geojson - and use the `Geometry` type.

File: src/pages/organisationUnits/form/GeometryFields.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                    value={input.value.coordinates?.[1].toString()}
+                    min="-180"
+                    max="180"
+                    step="any"
+                />
+            </Field>
+        </>
+    ) : (
+        <InputField
+            value={i18n.t('{{type}} coordinates are not editable', {
```
Comment (Birkbjo) [inline src/pages/organisationUnits/form/GeometryFields.tsx:68]: We should probably translate this by adding by `geometryType` object to `translatedModelConstants`, and use `type: getConstantTranslation: input.value?.type`

File: src/pages/organisationUnits/form/GeometryFields.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
+import i18n from '@dhis2/d2-i18n'
+import { Field, InputField } from '@dhis2/ui'
+import React from 'react'
+import { useField } from 'react-final-form'
+
+export function GeometryFields() {
+    const fieldName = 'geometry'
+    const { input, meta } = useField(fieldName)
+
+    const handleChange = ({
```
[the remark this answers is not in the document]
Comment (flaminic) [reply src/pages/organisationUnits/form/GeometryFields.tsx:10]: Makes complete sense

File: src/pages/organisationUnits/Edit.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    })
+
+    return (
+        <FormBase
+            onSubmit={useOnSubmitEdit({
+                section,
+                modelId,
+            })}
+            section={section}
+            initialValues={orgUnit.data as OrgUnitFormValues}
```
Comment (Birkbjo) [inline src/pages/organisationUnits/Edit.tsx:74]: `as` shouldn't be necessary here.

File: src/pages/organisationUnits/New.tsx [hunk 6]
```diff
@@ -3,41 +3,22 @@ import { FormBase } from '../../components'
...
 export const Component = () => {
     return (
         <FormBase
             onSubmit={useOnSubmitNew({
                 section,
-                valueFormatter: formatFormValues,
             })}
-            initialValues={initialValues as FormValues}
-            validate={(values: FormValues) => {
+            initialValues={initialValues as OrgUnitFormValues}
```
Comment (Birkbjo) [inline src/pages/organisationUnits/New.tsx:20]: `as` shouldn't be necessary if you remove ` as z.AnyZodObject` in `schema`. Type should be inferred from `getDefaults`.

File: src/pages/organisationUnits/New.tsx [hunk 7]
```diff
@@ -3,41 +3,22 @@ import { FormBase } from '../../components'
...
+            validate={(values: OrgUnitFormValues) => {
```
Comment (Birkbjo) [inline src/pages/organisationUnits/New.tsx:21]: type should be inferred from `initialValues` above. No need to type this (and it's not actually `OrgUnitFormValues` here - it should be `typeof initialValues`.

File: src/pages/organisationUnits/form/GeometryFields.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (flaminic) [reply src/pages/organisationUnits/form/GeometryFields.tsx:71]: @Birkbjo i need to understand what you mean here :)

**Review by Birkbjo — APPROVED** [review]

File: src/pages/organisationUnits/form/GeometryFields.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (flaminic) [reply src/pages/organisationUnits/form/GeometryFields.tsx:71]: i see :)

**Pull request merged**